app/src/main/java/bg/zahov/app/ui/settings/profile/EditProfileViewModel.kt

```diff
@@ -1,121 +1,160 @@
 package bg.zahov.app.ui.settings.profile
 
-import android.app.Application
-import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import bg.zahov.app.Inject
 import bg.zahov.app.data.exception.CriticalDataNullException
-import bg.zahov.app.getServiceErrorProvider
-import bg.zahov.app.getUserProvider
+import bg.zahov.app.data.interfaces.ServiceErrorHandler
+import bg.zahov.app.data.interfaces.UserProvider
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
-class EditProfileViewModel(application: Application) : AndroidViewModel(application) {
-    private val repo by lazy {
-        application.getUserProvider()
-    }
-    private val serviceErrorHandler by lazy {
-        application.getServiceErrorProvider()
-    }
-    private val _state = MutableLiveData<State>()
-    val state: LiveData<State>
-        get() = _state
+/**
+ * ViewModel responsible for handling user profile editing logic.
+ *
+ * @property repo The user provider for accessing and updating user information.
+ * @property errorHandler Handles service errors, particularly critical data null exceptions.
+ */
+class EditProfileViewModel(
+    private val repo: UserProvider = Inject.userProvider,
+    private val errorHandler: ServiceErrorHandler = Inject.serviceErrorHandler
+) : ViewModel() {
+
+    /**
+     * UI state for EditProfileViewModel, containing properties related to authentication,
+     * user credentials, password visibility, and notifications.
+     *
+     * @property authenticated Whether the user is authenticated.
+     * @property username The current username.
+     * @property password The current password.
+     * @property passwordVisibility Controls the visibility of the password.
+     * @property notify Optional notification message.
+     */
+    data class UiState(
+        val authenticated: Boolean = false,
+        val username: String = "",
+        val password: String = "",
+        val passwordVisibility: Boolean = false,
+        val notify: String? = null,
```

Comment:
rename it - userMessages, messages or something more explicit

Reply:
:thumbs up:

Reply:
still same name